test/integration/kubernetes_deploy_test.rb

```diff
@@ -632,6 +632,18 @@ def test_can_deploy_deployment_with_zero_replicas
     ])
   end
 
+  def test_deploy_successful_with_partial_availability
+    result = deploy_fixtures("slow-cloud")
+    assert_deploy_success(result)
+
+    result = deploy_fixtures("slow-cloud")
+    assert_deploy_success(result)
+
+    assert_logs_match_all(
+      [%r{Deployment\/web\s+[34] replicas, 3 updatedReplicas, 2 availableReplicas, [12] unavailableReplica}]
```

Comment:
This test still isn't that reliable. But I'm skeptical that increasing the sleep or using `initialDelaySeconds` is going to make it better. The rest of the feedback is addressed.

Reply:
The reason I think tuning the sleep/delay could help is that the gem only polls for statuses every 3 seconds. So e.g. if we make a scaling operation take, say, 5-7 seconds, we should have a much better chance of succeeding at a consistent point in the scaling, because we'll see every step.

If I'm wrong about that, or it makes the test hella slow, let's use a fancier regex (e.g. `Deployment\/web\s+(?<replicas>\d+) replicas?, (?<updatedReplicas>\d+) updatedReplicas?, (?<availableReplicas>\d+) availableReplicas?`) to parse the numbers we want out of the logs and make looser assertions (e.g. `availableReplicas < replicas`, `updatedReplicas < replicas`). It's shitty, but having multiple flaky tests is worse.

Reply:
The last two commits have passed all tests on the first pass, so I'd comfortable calling them reliable.

Reply:
I just ran this one 100 times locally, and it failed 3 times:
* `Deployment/web                                    3 replicas, 3 updatedReplicas, 3 availableReplicas`
* `Deployment/web                                    3 replicas, 3 updatedReplicas, 3 availableReplicas`
* `Deployment/web                                    4 replicas, 2 updatedReplicas, 3 availableReplicas, 1 unavailableReplica`

That's not _horrible_, but considering that there are three of them, they could end up noticeably increasing the fail rate overall. I'm thinking we should make sure we have good unit coverage of the `%` option and remove that copy of the test. Maybe the restart copy too. @stefanmb wdyt? Do you have any ideas for improving the reliability here?

Reply:
We could change the restart one to be `%` based and take out the `%` one here. That would save us a test w/o any loss of coverage.